pkg/util/proxyconfig/no_proxy.go

```diff
@@ -102,6 +102,15 @@ func MergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructur
 			// From https://cloud.google.com/vpc/docs/special-configurations add GCP metadata.
 			// "metadata.google.internal." added due to https://bugzilla.redhat.com/show_bug.cgi?id=1754049
 			set.Insert("metadata", "metadata.google.internal", "metadata.google.internal.")
+		case configv1.NutanixPlatformType:
+			// Add user's Prism Central endpoint to the set
+			prismCentral := infra.Spec.PlatformSpec.Nutanix.PrismCentral.Address
```

Comment:
The PC and PE endpoint addresses are in the Infrastructure CR's platformSpec, not dependent on the platformStatus. So I think we should move the code outside of this "if infra.Status.PlatformStatus != nil {" block, and put in a separate block in this function as below:

if infra.Spec.PlatformSpec.Type == configv1.NutanixPlatformType && infra.Spec.PlatformSpec.Nutanix != nil {
		// Add user's Prism Central endpoint to the set
		prismCentral := infra.Spec.PlatformSpec.Nutanix.PrismCentral.Address
		set.Insert(prismCentral)

		// Add user's Prism Element endpoints to the set
		for _, prismElement := range infra.Spec.PlatformSpec.Nutanix.PrismElements {
			set.Insert(prismElement.Endpoint.Address)
		}
	}